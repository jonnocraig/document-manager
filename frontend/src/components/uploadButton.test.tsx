import * as React from 'react';
import { shallow } from 'enzyme';
import UploadButton from './uploadButton';

function setup(nextProps = {}) {
  const props = {
    handleUpload: jest.fn(),
    ...nextProps
  };
  const wrapper = shallow(<UploadButton {...props} />);

  return {
    props,
    wrapper
  };
}

describe('<UploadButton />', () => {
  it('should match the snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should fire handeUpload if changed', () => {
    const { props, wrapper } = setup();
    const event = { preventDefault: jest.fn(), target: {files: [ 'test.txt' ]}};
    wrapper.find('#uploadDocument').simulate('change', event);
    expect(props.handleUpload.mock.calls.length).toBe(1);    
  });
});
