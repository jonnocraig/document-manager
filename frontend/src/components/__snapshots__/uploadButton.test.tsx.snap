// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadButton /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UploadButton
    handleUpload={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <FormGroup
        tag="div"
      >
        <InputGroup
          tag="div"
        >
          <div
            className="file-input-wrapper"
          >
            <Input
              id="uploadDocument"
              name="file"
              onChange={[MockFunction]}
              type="file"
            />
            <Button
              color="success"
              tag="button"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="file-upload"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
               Upload
            </Button>
          </div>
        </InputGroup>
      </FormGroup>,
      "tag": "form",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <InputGroup
          tag="div"
        >
          <div
            className="file-input-wrapper"
          >
            <Input
              id="uploadDocument"
              name="file"
              onChange={[MockFunction]}
              type="file"
            />
            <Button
              color="success"
              tag="button"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="file-upload"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
               Upload
            </Button>
          </div>
        </InputGroup>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            className="file-input-wrapper"
          >
            <Input
              id="uploadDocument"
              name="file"
              onChange={[MockFunction]}
              type="file"
            />
            <Button
              color="success"
              tag="button"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="file-upload"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
               Upload
            </Button>
          </div>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Input
                id="uploadDocument"
                name="file"
                onChange={[MockFunction]}
                type="file"
              />,
              <Button
                color="success"
                tag="button"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="file-upload"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                 Upload
              </Button>,
            ],
            "className": "file-input-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "uploadDocument",
                "name": "file",
                "onChange": [MockFunction],
                "type": "file",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon="file-upload"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />,
                  " Upload",
                ],
                "color": "success",
                "tag": "button",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": "file-upload",
                    "inverse": false,
                    "listItem": false,
                    "mask": null,
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "title": "",
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                " Upload",
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <FormGroup
          tag="div"
        >
          <InputGroup
            tag="div"
          >
            <div
              className="file-input-wrapper"
            >
              <Input
                id="uploadDocument"
                name="file"
                onChange={[MockFunction]}
                type="file"
              />
              <Button
                color="success"
                tag="button"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="file-upload"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                 Upload
              </Button>
            </div>
          </InputGroup>
        </FormGroup>,
        "tag": "form",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <InputGroup
            tag="div"
          >
            <div
              className="file-input-wrapper"
            >
              <Input
                id="uploadDocument"
                name="file"
                onChange={[MockFunction]}
                type="file"
              />
              <Button
                color="success"
                tag="button"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="file-upload"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                 Upload
              </Button>
            </div>
          </InputGroup>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div
              className="file-input-wrapper"
            >
              <Input
                id="uploadDocument"
                name="file"
                onChange={[MockFunction]}
                type="file"
              />
              <Button
                color="success"
                tag="button"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="file-upload"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                 Upload
              </Button>
            </div>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Input
                  id="uploadDocument"
                  name="file"
                  onChange={[MockFunction]}
                  type="file"
                />,
                <Button
                  color="success"
                  tag="button"
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon="file-upload"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />
                   Upload
                </Button>,
              ],
              "className": "file-input-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "id": "uploadDocument",
                  "name": "file",
                  "onChange": [MockFunction],
                  "type": "file",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <FontAwesomeIcon
                      border={false}
                      className=""
                      fixedWidth={false}
                      flip={null}
                      icon="file-upload"
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size={null}
                      spin={false}
                      symbol={false}
                      title=""
                      transform={null}
                    />,
                    " Upload",
                  ],
                  "color": "success",
                  "tag": "button",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "border": false,
                      "className": "",
                      "fixedWidth": false,
                      "flip": null,
                      "icon": "file-upload",
                      "inverse": false,
                      "listItem": false,
                      "mask": null,
                      "pull": null,
                      "pulse": false,
                      "rotation": null,
                      "size": null,
                      "spin": false,
                      "symbol": false,
                      "title": "",
                      "transform": null,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  " Upload",
                ],
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
